.github/workflows/szdiff.yml

```diff
@@ -0,0 +1,53 @@
+name: Check Line Counts
+on:
+  pull_request_target:
+
+# Cancel the workflow in progress in newer build is about to start.
+concurrency:
+  group: ${{ github.workflow }}-${{ github.head_ref || github.run_id }}
+  cancel-in-progress: true
+
+jobs:
+  szdiff:
+    name: Core Library Line Difference
+    permissions:
+      contents: read
+      pull-requests: write
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code from PR branch
+        uses: actions/checkout@v4
+        with:
+          repository: ${{ github.event.pull_request.head.repo.full_name }}
+          ref: ${{ github.event.pull_request.head.sha }}
+          path: pr
+        # the base must be from tinygrad master for security purpose
+      - name: Checkout code from tinygrad master
+        uses: actions/checkout@v4
+        with:
+          repository: tinygrad/tinygrad
+          ref: master
+          path: base
```

Comment:
doesn't actions/checkout default to the base branch when its `pull_request_target`?

Reply:
Just did a quick test, you are correct.
![image](https://github.com/tinygrad/tinygrad/assets/17937555/7f4f9a96-6126-41e7-9e37-ab0427751b8d)